src/subcommands/top.ts

```diff
@@ -0,0 +1,168 @@
+`;
+
+export default async function topSubcommand(args: Args) {
+  if (args.help) {
+    console.log(help);
+    Deno.exit(0);
+  }
+  if (!args.project) {
+    error(
+      "No project specified. Use --project to specify the project of which to stream the stats",
+    );
+  }
+  let format: Format;
+  switch (args.format) {
+    case "table":
+    case "json":
+      format = args.format;
+      break;
+    case undefined:
+      format = Deno.isatty(Deno.stdout.rid) ? "table" : "json";
+      break;
+    default:
+      error(
+        `Invalid format '${args.format}'. Supported values for the --format option are 'table' or 'json'`,
+      );
+  }
+
+  const spinner = wait(
+    `Connecting to the stats stream of project '${args.project}'...`,
+  ).start();
+  const api = args.token
+    ? API.fromToken(args.token)
+    : API.withTokenProvisioner(TokenProvisioner);
+  let stats;
+  try {
+    stats = await api.stream_metering(args.project!);
+  } catch (err) {
+    spinner.fail(
+      `Failed to connect to the stats stream of project '${args.project}': ${err.message}`,
+    );
+    return Deno.exit(1);
+  }
+  spinner.succeed(
+    `Connected to the stats stream of project '${args.project}'`,
+  );
+  switch (format) {
+    case "table":
+      return await tabbed(stats);
+    case "json":
+      return await json(stats);
+  }
+}
+
+async function tabbed(stats: AsyncGenerator<ProjectStats, void>) {
+  const table: { [id: string]: unknown } = {};
+  const timeouts: { [id: string]: number } = {};
+  const toDelete: string[] = [];
+  const spinner = wait("Streaming...").start();
+  try {
+    let next = stats.next();
+    while (true) {
+      const result = await Promise.race([next, delay(10_000)]);
+      const previousLength = Object.keys(table).length;
+      if (result) {
+        next = stats.next();
+        const stat = result.value;
+        if (!stat) {
+          spinner.succeed("Stream ended");
+          return;
+        }
+        const id = encodeHex(await sha256(stat.id + stat.region))
+          .slice(0, 6);
+        table[id] = {
+          "region": stat.region,
+          "Req/min": Math.ceil(stat.requestsPerMinute),
+          "CPU%": parseFloat((stat.cpuTimePerSecond / 10).toFixed(2)),
+          "CPU/req": parseFloat((stat.cpuTimePerRequest || 0).toFixed(2)),
+          "RSS/5min": parseFloat(
+            (stat.maxRss5Minutes / 1_000_000).toFixed(3),
+          ),
+          "Ingress/min": parseFloat(
+            (stat.ingressBytesPerMinute / 1_000).toFixed(3),
+          ),
+          "Egress/min": parseFloat(
+            (stat.egressBytesPerMinute / 1_000).toFixed(3),
+          ),
+          "KVr/min": Math.ceil(stat.kvReadUnitsPerMinute),
+          "KVw/min": Math.ceil(stat.kvWriteUnitsPerMinute),
+          "QSenq/min": Math.ceil(stat.enqueuePerMinute),
+          "QSdeq/min": Math.ceil(stat.dequeuePerMinute),
+        };
```

Comment:
Any reason to use `unknown` type as the value type of `table`?
It looks like the type is determined statically in L111-L129

Reply:
This is the final data outputed to stdout; nothing is expected to use these objects, only print them to the console. As such, I think unknown is a fair choice, as it correctly conveys this. Having to define the interface, and consequently having to change the headers of the table in 2 places instead of just one is unnecessary work.